File: algorithms/algorithms-cplusplus/leetcode/Question_0209_Minimum_Size_Subarray_Sum.cc
#include <limits.h>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
 public:
  int minSubArrayLen(int s, vector<int>& nums) {
    int res = INT_MAX, sum = 0;
    for (int i = 0, j = 0; j < nums.size(); j++) {
      sum += nums[j];
      while (sum >= s) {
        res = min(res, j - i + 1);
        sum -= nums[i++];
      }
    }
    return (res == INT_MAX) ? 0 : res;
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0209_Minimum_Size_Subarray_Sum.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0209_Minimum_Size_Subarray_Sum {
    public int minSubArrayLen(int s, int[] nums) {
        int res = Integer.MAX_VALUE, sum = 0;
        for (int i = 0, j = 0; j < nums.length; j++) {
            sum += nums[j];
            while (sum >= s) {
                res = Math.min(res, j - i + 1);
                sum -= nums[i++];
            }
        }
        return (res == Integer.MAX_VALUE) ? 0 : res;
    }
}
